sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -469,6 +469,13 @@ trait FileFormat {
       options: Map[String, String]): RDD[InternalRow]
 
   /**
+   * Returns whether this format support returning columnar batch or not.
```

Comment:
Maybe a TODO, that we should just have different traits for the different formats.